kutt.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/bugsnag/microkit/clog"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+type Client struct {
+	HTTPClient *http.Client
+	ApiKey     string
+	BaseURL    string
+	UserAgent  string
+}
+
+type SubmitParams struct {
+	URL      string `json:"target"`
+	ExpireIn string `json:"expire_in"`
+}
+
+func NewClient() *Client {
+	var cli Client
+	cli.ApiKey = viper.GetString("kutt_api_key")
+	cli.BaseURL = viper.GetString("kutt_base_url")
+	cli.UserAgent = "promalert/v1 (+https://github.com/bugsnag/promalert)"
+
+	return &cli
+}
+
+func (cli *Client) httpClient() *http.Client {
+	if cli.HTTPClient != nil {
+		return cli.HTTPClient
+	}
+	return http.DefaultClient
+}
+
+func (cli *Client) error(statusCode int, body io.Reader) error {
+	buf, err := ioutil.ReadAll(body)
+	if err != nil || len(buf) == 0 {
+		return errors.Errorf("request failed with status code %d", statusCode)
+	}
+	return errors.Errorf("StatusCode: %d, Error: %s", statusCode, string(buf))
+}
+
+func (cli *Client) do(req *http.Request) (*http.Response, error) {
+	req.Header.Set("X-API-Key", cli.ApiKey)
+	req.Header.Set("User-Agent", cli.UserAgent)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	return cli.httpClient().Do(req)
+}
+
+func (cli *Client) Submit(target string) (*URL, error) {
+	path := "/api/url/submit"
+	reqURL := cli.BaseURL + path
```

Comment:
You could replace this with a sprintf:
```suggestion
	reqURL := fmt.Sprintf("%s/%s, cli.BaseURL, "api/url/submit")
```